### ISSUE TITLE
Rename ProgressBar callback to TQDMProgressBar

### ISSUE THREAD

**Issue by daniellepintz** [PyTorchLightning/pytorch-lightning#9302]
## Proposed refactoring or deprecation

Rename ProgressBar callback to TQDMProgressBar (https://github.com/tqdm/tqdm)
https://github.com/daniellepintz/pytorch-lightning/blob/a49690dfa3502508c8765e6bb46ded17e561af8e/pytorch_lightning/callbacks/progress/progress.py#L55-L55

### Motivation

After we added the RichProgressBar in https://github.com/PyTorchLightning/pytorch-lightning/pull/8929, it makes sense to rename ProgressBar to something more specific. 

### Pitch

Update all references of ProgressBar to TQDMProgressBar

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

**Comment by daniellepintz** [comment]
Could someone please add the `good first issue` label? Also could someone advise on how we usually go about making these kinds of changes backwards compatible? Or is this a breaking change?

**Comment by tchaton** [comment]
This is a breaking change and would have to go through depreciation cycle.

**Comment by carmocca** [comment]
If we do the following, it wouldn't be breaking, right?

```python
class TQDMProgressBar(ProgressBarBase):
    ...

# points to the default implementation
ProgressBar = TQDMProgressBar
```

**Comment by daniellepintz** [comment]
@carmocca yeah makes sense to me! looks like @kaushikb11 's PR fixes this

**Comment by InCogNiTo124** [comment]
I'd like to take on this, would you please assing this to me? @daniellepintz

**Comment by daniellepintz** [comment]
Thanks @InCogNiTo124 ! I think @kaushikb11 already has a PR out for this #9229 but hasnt merged it yet. @kaushikb11 what is the status of the PR?
